app/Livewire/ListIssues.php

```diff
@@ -102,11 +102,17 @@ public function mount(): void
 
     public function render(): View
     {
+        $this->initialiseLabels();
+
         $issues = $this->originalIssues
             ->filter(fn (Issue $issue): bool => $this->showIgnoredIssues === in_array($issue->url, $this->ignoredUrls, true))
             ->when($this->searchTerm, $this->applySearch())
             ->when($this->sort, $this->applySort());
 
+        foreach ($issues as $issue) {
+            $this->incrementLabelCounts($issue);
```

Comment:
This is a nice refactoring, but now the issues go through two loops. When search is applied, it goes through three loops. My original solution aimed to utilize the existing loop since, even without pagination, the loop can impact performance. Perhaps I might be overthinking it 😊.